fanficfare/adapters/adapter_royalroadcom.py

```diff
@@ -166,12 +166,12 @@ def extractChapterUrlsAndMetadata(self):
 
 
         chapters = soup.find('table',{'id':'chapters'}).find('tbody')
-        tds = [tr.findAll('td')[0] for tr in chapters.findAll('tr')]
+        tds = [tr.findAll('td') for tr in chapters.findAll('tr')]
         # Links in the RR ToC page are in the normalized long form, so match is simpler than in normalize_chapterurl()
         chap_pattern_long = r"https?://(?:www\.)?royalroadl?\.com/fiction/\d+/[^/]+/chapter/(\d+)/[^/]+/?$"
-        for td in tds:
-            chapterUrl = 'https://' + self.getSiteDomain() + td.a['href']
-            if self.add_chapter(td.text, chapterUrl):
+        for chapter,date in tds:
+            chapterUrl = 'https://' + self.getSiteDomain() + chapter.a['href']
+            if self.add_chapter(chapter.text, chapterUrl, { "date": self.make_date(date) }):
```

Comment:
Passing the chapter date is fine, but it should be string formatted.  See [adapter_archiveofourownorg.py#L270](https://github.com/JimmXinu/FanFicFare/blob/main/fanficfare/adapters/adapter_archiveofourownorg.py#L270)